domehunter/__init__.py

```diff
@@ -86,70 +230,291 @@ def is_home(self):
+            self.home_sensor_pin.drive_low()
+        # update the last_direction instance variable
+        self.last_direction = self.current_direction
+        # now update the current_direction variable to CW
         self.current_direction = "CW"
+        # set the direction relay switch to CW position
+        self.direction_relay.on()
+        # update the debug LEDs (will only do something if they are enabled)
+        if self.last_direction == "CCW":
+            self._turn_led_off(leds=['relay_2_nc'])
+        self._turn_led_on(leds=['relay_2_no'])
+        # turn on rotation
+        self.rotation_relay.on()
+        # update the rotation relay debug LEDs
+        self._turn_led_on(leds=['relay_1_no'])
+        self._turn_led_off(leds=['relay_1_nc'])
         cmd_status = True
         return cmd_status
 
-    def _move_ccw(self, degrees):
-        print(f"Sending GPIO move_ccw({degrees}) command.")
+    def _move_ccw(self):
+        """Set dome to move counter-clockwise."""
+        # if testing, deactivate the home_sernsor_pin to simulate leaving home
+        if self.testing and self._at_home:
+            self.home_sensor_pin.drive_low()
+        # update the last_direction instance variable
+        self.last_direction = self.current_direction
+        # now update the current_direction variable to CCW
         self.current_direction = "CCW"
+        # set the direction relay switch to CCW position
+        self.direction_relay.off()
+        # update the debug LEDs (will only do something if they are enabled)
+        if self.last_direction == "CW":
+            self._turn_led_off(leds=['relay_2_no'])
+        self._turn_led_on(leds=['relay_2_nc'])
+        # turn on rotation
+        self.rotation_relay.on()
+        # update the rotation relay debug LEDs
+        self._turn_led_on(leds=['relay_1_no'])
+        self._turn_led_off(leds=['relay_1_nc'])
         cmd_status = True
         return cmd_status
+
+    def _stop_moving(self):
+        """Stop dome movement by switching the dome rotation relay off."""
+        self.rotation_relay.off()
+        # update the debug LEDs
+        self._turn_led_off(leds=['relay_1_no'])
+        self._turn_led_on(leds=['relay_1_nc'])
+        # update last_direction with current_direction at time of method call
+        self.last_direction = self.current_direction
+
+    def _simulate_ticks(self, num_ticks):
+        """Method to simulate encoder ticks while in testing mode."""
+        tick_count = 0
+        # repeat this loop of driving the mock pins low then high to simulate
+        # an encoder tick. Continue until desired number of ticks is reached.
+        while tick_count < num_ticks:
+            self.encoder_pin.drive_low()
+            # t_wait is set so that it will always exceed the set bounce_time
+            # of the pins
+            time.sleep(self.t_wait)
+            self.encoder_pin.drive_high()
+            time.sleep(self.t_wait)
+            tick_count += 1
```

Comment:
Shouldn't this be incrementing the global `self.encoder_count`?

Reply:
Yeah, shouldn't this be using `_increment_count()`?

Reply:
OK, I don't get the need to have `encoder_pin` and `encoder` in testing mode and normal mode, respectively. Just makes it confusing. Ditto for `home_sensor_pin` and `home_sensor`. Let's chat about this one.